server/keyshare/tasks/tasks.go

```diff
@@ -115,40 +140,65 @@ func (t *taskHandler) sendExpiryEmails(ctx context.Context, id int64, username,
 				email,
 				lang,
 			)
-			return err
+
+			if t.revalidateMail && err != nil && err != keyshare.ErrNoNetwork {
+				// When email revalidation is enabled and sending was impossible because of
+				// (temporary) MX / A record issues at the domain or an invalid email address,
+				// we mark the record to be revalidated in 5 days from now.
+				if err = t.db.ExecUserContext(ctx, "UPDATE irma.emails SET revalidate_on = $1 WHERE id = $2",
+					time.Now().AddDate(0, 0, 5).Unix(),
+					id); err != nil {
+					t.conf.Logger.WithField("error", err).Error("could not update email address to set revalidate_on")
+					return err
+				}
+				return nil
+			} else {
+				return err
+			}
 		},
 		id,
 	)
 	if err != nil {
-		t.conf.Logger.WithField("error", err).Error("Could not retrieve user's email addresses")
+		t.conf.Logger.WithField("error", err).Error("could not retrieve user's email addresses")
 		return err
 	}
 	return nil
 }
 
-// Mark old unused accounts for deletion, and inform their owners.
+// Mark old unused accounts for deletion, and inform their owners. When email revalidation enabled,
+// email addresses which where marked for revalidation are skipped because these will be processed
+// separately inside revalidateEmails.
 func (t *taskHandler) expireAccounts(ctx context.Context) {
 	// Disable this task when email server is not given
 	if t.conf.EmailServer == "" {
-		t.conf.Logger.Warning("Expiring accounts is disabled, as no email server is configured")
+		t.conf.Logger.Warning("expiring accounts is disabled, as no email server is configured")
```

Comment:
Logging may be capitalised. The requirement that errors should be lowercase only holds for go errors.